### PR TITLE
Restore 1.7 changes for `material.ExposedDropdownMenu`

### PR DESCRIPTION
**Pull request by MatkovIvan**
Apply missed diff from 1.7 AOSP to a commonized version. Mostly it's about formatting, but also it changes `interface` to `abstract class`

Source of changes is f5541f29d045c6ba9734689ec67891f8d667412b (androidx/compose-material3/1.3.0-beta03)

Fixes [CMP-1247](https://youtrack.jetbrains.com/issue/CMP-1247), [CMP-1248](https://youtrack.jetbrains.com/issue/CMP-1248)

### CODE REVIEW

File: compose/material/material/src/commonMain/kotlin/androidx/compose/material/ExposedDropdownMenu.kt [hunk 1]
```diff
@@ -132,63 +121,49 @@ internal expect fun ExposedDropdownMenuBoxScope.ExposedDropdownMenuDefaultImpl(
     content: @Composable ColumnScope.() -> Unit
 )
 
-internal fun Modifier.expandable(
-    onExpandedChange: () -> Unit,
-    menuLabel: String
-) = composed {
-    val currentOnExpandedChange by rememberUpdatedState(onExpandedChange)
-    pointerInput(Unit) {
+internal fun Modifier.expandable(onExpandedChange: () -> Unit, menuLabel: String) =
```
Comment (igordmn) [inline compose/material/material/src/commonMain/kotlin/androidx/compose/material/ExposedDropdownMenu.kt:124]: Please, specify the commit in the PR description, which was chosen as a base.

I looked at the last merged `androidx/compose-ui/1.7.0-beta03` tag, file `compose/material/material/src/androidMain/kotlin/androidx/compose/material/ExposedDropdownMenu.android.kt`. It contains:
```
private fun Modifier.expandable(
    onExpandedChange: () -> Unit,
    menuLabel: String
) = pointerInput(onExpandedChange) {
```

Which doesn't correspond this formatting.

It probably means that some later commit from `androidx-main` was chosen as a base. It is probably fine if API is the same, but we should be more careful.

Comment (MatkovIvan) [reply compose/material/material/src/commonMain/kotlin/androidx/compose/material/ExposedDropdownMenu.kt:124]: Aha. Found why - I used beta03 commit, but from material3 🙃

I've rechecked APIs - it's the same. My suggestion is to keep it as is to simplify future merges

Comment (igordmn) [reply compose/material/material/src/commonMain/kotlin/androidx/compose/material/ExposedDropdownMenu.kt:124]: Sure, but let's add this commit to the description. Is it `androidx/compose-material3/1.3.0-beta03`?

Comment (MatkovIvan) [reply compose/material/material/src/commonMain/kotlin/androidx/compose/material/ExposedDropdownMenu.kt:124]: Done. Yes

**Review by igordmn — APPROVED** [review]

**Pull request merged**